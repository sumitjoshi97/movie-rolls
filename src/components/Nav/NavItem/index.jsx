import React from 'react'
import PropTypes from 'prop-types'

const NavItem = ({ index, name, type, isActive, ...props }) => {
  const style = {
    active: {
      borderBottom: '0.5rem solid #ff1e1e'
    },
    inActive: {
      borderBottom: '0.5rem solid transparent'
    }
  }

  return (
    <div
      name={name}
      className='nav-item'
      style={isActive ? style.active : style.inActive}
      onClick={() => {
        props.setActive(index)
        props.handleNav(type)
      }}
    >
      {name}
    </div>
  )
}

NavItem.propTypes = {
  index: PropTypes.number,
  name: PropTypes.string,
  type: PropTypes.string,
  isActive: PropTypes.bool,
  setActive: PropTypes.func,
  handleNav: PropTypes.func
}

export default NavItem
