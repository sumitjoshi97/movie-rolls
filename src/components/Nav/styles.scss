@import "../../sass/variables";
@import "../../sass/mixin";

.nav {
  position: absolute;
  width: 100%;
  bottom: 0rem;
  display: flex;
  flex-direction: row;
  justify-content: center;

  & > * {
    text-transform: uppercase;
    margin: 0 $padding-large;
    @include respond(phone) {
      margin: 0 4rem;
    }
    cursor: pointer;
  }
}

.nav-item {
  padding-bottom: 1rem;
  letter-spacing: 2px;
}
