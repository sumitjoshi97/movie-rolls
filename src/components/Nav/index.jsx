import React, { Component } from 'react'

import './styles.scss'

export default class Nav extends Component {
  render() {
    return (
      <nav className='nav'>
        <div className='nav__movies'>Movies</div>
        <div className='nav__shows'>TV shows</div>
      </nav>
    )
  }
}
