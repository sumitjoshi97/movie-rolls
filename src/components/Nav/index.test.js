import React from 'react'
import { shallow } from 'enzyme'

import Nav from './index'
import NavItem from './NavItem'

describe('Nav component unit tests', () => {
  const options = [
    {
      type: 'type1',
      name: 'type1'
    },
    {
      type: 'type2',
      name: 'type2'
    }
  ]

  const handleNav = jest.fn()

  const props = { options, handleNav }

  const navWrapper = shallow(<Nav {...props} />)

  test('should render properly', () => {
    expect(navWrapper).toMatchSnapshot()
  })

  test('should render 2 nav-items', () => {
    expect(navWrapper.find(NavItem)).toHaveLength(2)
  })
})
