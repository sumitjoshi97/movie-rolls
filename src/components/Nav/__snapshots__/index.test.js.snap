// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component unit tests should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    handleNav={[MockFunction]}
    options={
      Array [
        Object {
          "name": "type1",
          "type": "type1",
        },
        Object {
          "name": "type2",
          "type": "type2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavItem
          handleNav={[MockFunction]}
          index={0}
          isActive={true}
          name="type1"
          setActive={[Function]}
          type="type1"
        />,
        <NavItem
          handleNav={[MockFunction]}
          index={1}
          isActive={false}
          name="type2"
          setActive={[Function]}
          type="type2"
        />,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "type1",
        "nodeType": "function",
        "props": Object {
          "handleNav": [MockFunction],
          "index": 0,
          "isActive": true,
          "name": "type1",
          "setActive": [Function],
          "type": "type1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "type2",
        "nodeType": "function",
        "props": Object {
          "handleNav": [MockFunction],
          "index": 1,
          "isActive": false,
          "name": "type2",
          "setActive": [Function],
          "type": "type2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavItem
            handleNav={[MockFunction]}
            index={0}
            isActive={true}
            name="type1"
            setActive={[Function]}
            type="type1"
          />,
          <NavItem
            handleNav={[MockFunction]}
            index={1}
            isActive={false}
            name="type2"
            setActive={[Function]}
            type="type2"
          />,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "type1",
          "nodeType": "function",
          "props": Object {
            "handleNav": [MockFunction],
            "index": 0,
            "isActive": true,
            "name": "type1",
            "setActive": [Function],
            "type": "type1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "type2",
          "nodeType": "function",
          "props": Object {
            "handleNav": [MockFunction],
            "index": 1,
            "isActive": false,
            "name": "type2",
            "setActive": [Function],
            "type": "type2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
